libsolidity/formal/SMTChecker.cpp

```diff
@@ -875,3 +865,14 @@ void SMTChecker::addPathImpliedExpression(smt::Expression const& _e)
 {
 	m_interface->addAssertion(smt::Expression::implies(currentPathConditions(), _e));
 }
+
+SMTChecker::VariableSequenceCounters SMTChecker::copyVariables() const
+{
+	VariableSequenceCounters vars;
+	for (auto var : m_variables)
+	{
+		if (SymbolicIntVariable *sVar = dynamic_cast<SymbolicIntVariable*>(var.second.get()))
```

Comment:
You can also implement a virtual `copy` function.

Reply:
Another solution would be to decouple the sequence counting and the type-dependent parts in the following way:

The first part is implemented in a generic class that holds a shared pointer to the specific type-dependent virtual implementations. If you copy the object (using just plain C++ assignment), it only copies the first part, containing the current sequence counter. The second part is - as well as the "next free sequence number" is behind a shard pointer and thus unique per variable and also able to do virtual function calls properly.

Reply:
I quite like the second idea. It's now implemented in that way.